### PR TITLE
Feature/extend isbn finding

### PR DESCRIPTION
**Pull request by mekarpeles**
@finnless's work w/ ISBN detection

### CODE REVIEW

File: bgp/__init__.py [hunk 1]
```diff
@@ -56,7 +56,10 @@
 def _memoize_xml(self):
     if not hasattr(self, '_xml'):
         _memoize_xml_tic = time.perf_counter()
-        self._xml = self.download(formats=['Djvu XML'], return_responses=True)[0].text
+        try:
+            self._xml = self.download(formats=['Djvu XML'], return_responses=True)[0].text
+        except requests.exceptions.Timeout as e:
+            logging.error('Timeout for xml for item - ' + sq.book.identifier + ' | ' + e)
```
Comment (mekarpeles) [inline bgp/__init__.py:62]: What is `sq` here?

Comment (mekarpeles) [reply bgp/__init__.py:62]: I think this is supposed to be the sequence (https://github.com/Open-Book-Genome-Project/sequencer/pull/58/files#diff-a0ba19faf07afcd67bbf3609db4c8b0f0d5f3ca73ca23a9e2c1e82d1fe300cd9R159), but I don't think it's in scope

File: bgp/__init__.py [hunk 2]
```diff
@@ -171,9 +182,9 @@ def sequence(self, book):
         'term_freq': WordFreqModule(),
         'urls': UrlExtractorModule()
     }, n=1, stop_words=None),
-    'fulltext': FulltextProcessor(modules={
-        'readinglevel': ReadingLevelModule()
-    }),
+    #'fulltext': FulltextProcessor(modules={
+    #    'readinglevel': ReadingLevelModule()
+    #}),
```
Comment (mekarpeles) [inline bgp/__init__.py:187]:
```suggestion
    'fulltext': FulltextProcessor(modules={
        'readinglevel': ReadingLevelModule()
    }),
```

File: bgp/modules/terms.py [hunk 3]
```diff
@@ -325,18 +334,31 @@ class CopyrightPageDetectorModule(KeywordPageDetectorModule):
...
+            if 'sbn' in word.lower():
                 in_isbn = True
+        else:
+            isbn = IsbnExtractorModule.validate_isbn(candidate_isbn)
+            if isbn:
+                isbns.append(isbn)
 
         return {
             'isbns': isbns,
+            #'page':  " ".join(word.text for word in page.iter('WORD'))
```
Comment (mekarpeles) [inline bgp/modules/terms.py:361]:
```suggestion
```

File: bgp/__init__.py (hunk 1 above)
Comment (finnless) [reply bgp/__init__.py:62]:
```suggestion
            logging.error('Timeout for xml for item - ' + self.identifier + ' | ' + e)
```

File: bgp/__init__.py [hunk 4]
```diff
@@ -67,7 +70,10 @@ def _memoize_plaintext(self):
     if hasattr(self, 'xml'):
         if not hasattr(self, '_plaintext'):
             _memoize_plaintext_tic = time.perf_counter()
-            self._plaintext = self.download(formats=['DjVuTXT'], return_responses=True)[0].text
+            try:
+                self._plaintext = self.download(formats=['DjVuTXT'], return_responses=True)[0].text
+            except requests.exceptions.Timeout as e:
+                logging.error('Timeout for txt for item - ' + sq.book.identifier + ' | ' + e)
```
Comment (finnless) [inline bgp/__init__.py:76]:
```suggestion
                logging.error('Timeout for txt for item - ' + self.identifier + ' | ' + e)
```

**Review by finnless — CHANGES_REQUESTED** [review]
This is the correct scope.

**Review by finnless — APPROVED** [review]

**Pull request merged**